lib/tx.py

```diff
@@ -217,3 +217,11 @@ def read_tx(self):
 
         return TxSegWit(version, marker, flag, inputs,
                         outputs, witness, locktime), double_sha256(orig_ser)
+
+
+class DeserializerFairCoin(Deserializer):
+    def read_block(self):
+        '''Returns a list of (deserialized_tx, tx_hash) pairs.'''
+        read_tx = self.read_tx
+        txs = [read_tx() for n in range(self._read_varint())]
+        return txs
```

Comment:
Oh, and please add a comment where the assertion used to be that some coins have excess data beyond the end of the transactions.  Thanks.